### PR TITLE
Simplify tests with jest

### PR DESCRIPTION
**Pull request by pavadeli**
Simplifies codebase and solves some issues with recent versions of nyc and mocha. Slows down tests from <1s to >5s, but that is not an issue.

### CODE REVIEW

File: extensions/sherlock-utils/src/control-flow.ts [hunk 1]
```diff
@@ -130,8 +130,14 @@ function prepareOptions<V>(base: Derivable<V>, opts?: ControlFlowOptions<V>, ):
     const result: PreparedOptions<V> = {};
     if (opts) {
         for (const key of Object.keys(opts) as Array<keyof typeof opts>) {
-            const opt = opts[key];
-            result[key] = typeof opt === 'function' ? derive(() => unwrap(opt(base))) : opt;
+            switch (key) {
```
Comment (njirem) [inline extensions/sherlock-utils/src/control-flow.ts]: Normally I'd say that this would be a separate PR

**Comment by pavadeli** [comment]
> Seems like a good change. But it actually is 4 PR's in one.
> 
> * Update dependencies
> * ~Fix bug~
> * Replace mocha/chai with jest
> * Rename all the .spec files
> 
> With that in mind, have you considered the implications of that last change, for the currently open branches? Especially the `tutorial` branch, which is completely based on mocha/chai. Granted, jest does have some advantages for the tutorial, but it is till quite a pain to rewrite it at this stage.
> 
> Why have you renamed `.spec` to `.test`, the function of the files hasn't changed. It should still both spec and test the code. But from a reviewing perspective, it makes everything a lot harder. Since some files are not detected as 'moves' but deletes+adds.

You are absolutely right! (Except the _fix bug_ bullet; the change you referred to was only to satisfy the latest TypeScript strictness) I will help with converting the tutorial to Jest.
